src/target/riscv/riscv-013.c

```diff
@@ -907,12 +907,12 @@ static riscv_reg_t read_abstract_arg(struct target *target, unsigned index,
 			LOG_TARGET_ERROR(target, "Unsupported size: %d bits", size_bits);
 			return ~0;
 		case 64:
-			dm_read(target, &v, DM_DATA0 + offset + 1);
-			value |= ((uint64_t) v) << 32;
+			if (dm_read(target, &v, DM_DATA0 + offset + 1) == ERROR_OK)
+				value |= ((uint64_t)v) << 32;
 			/* falls through */
 		case 32:
-			dm_read(target, &v, DM_DATA0 + offset);
-			value |= v;
+			if (dm_read(target, &v, DM_DATA0 + offset) == ERROR_OK)
+				value |= v;
```

Comment:
I don't think this is the way to fix the issue. It will not be possible to distinguish between reading a zero and reading nothing due to an error. I would suggest changing the signature of the function:
```
static int read_abstract_arg(..., riscv_reg_t *value)
```
Although this would create a much bigger diff -- a separate patch perhaps would be a better solution.

If you wish to leave this as-is for now, please, add a `FIXME` comment.

Reply:
I agree it would be better to pass up the error code, but I'm just trying to fix the compiler warnings, not everything that is wrong with the code. I filed #981 to track this.

Reply:
I also agree with @en-sc that this should (eventually) be fixed differently - by passing the error code up.

Thank you for filling the #981 to address this.